realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -462,6 +462,25 @@ public void setEmbedded(boolean embedded) {
         }
     }
 
+    /**
+     * Checks whether a given class property can represent a valid embedded object
+     * {@code @LinkingObjects} backlink that points to a parent. Valid embedded object backlink
+     * typs are {@link RealmFieldType#OBJECT} and {@link RealmFieldType#LIST}.
+     *
+     * @param linkingProperty the field for which we want to know the type.
+     * @return the name of the property.
+     * @throws IllegalStateException if the given property is not found in the schema or the
+     * backlink property is not a valid type.
+     */
+    public abstract String getLinkedType(String linkingProperty);
```

Comment:
Since this is a new method in the public API we should also have tests for this directly instead of indirectly through other methods.

Reply:
I have re-thought this method and it can be for internal use only.

Reply:
👍